### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 23

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#263] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152317021-69758e33-e027-4aec-9fd9-4f8ec26c7eb0.png)

**Comment by LuginiAndrea** [comment]
La prima è falsa
$$f: \mathbf{Z} \rightarrow \mathbf{Z}: f\left(x\right) = x + 1$$
$$g:\mathbf{Z} \rightarrow \mathbf{N}: g\left(z\right) = z^2$$
Nonostante siano rispettati tutti i vincoli, la prima è iniettiva, la seconda no

La seconda è falsa, basti infatti considerare una $f$ non iniettiva quale 
$$f:\mathbf{Z} \rightarrow \mathbf{N}: f\left(x\right) = 2$$
$$f\left(\mathbf{Z}\right) - f\left(\mathbf{N}\right) = 2 - 2 = \emptyset$$
$$f\left(\mathbf{Z} - \mathbf{N}\right) = f\left(-\mathbf{N}\right) = 2 \neq \emptyset \Longrightarrow f\left(\mathbf{Z} - \mathbf{N}\right) \neq f\left(\mathbf{Z}\right) - f\left(\mathbf{N}\right)$$

Infine anche la terza è falsa, non essendo implicata nè la suriettività di $f$ nè la suriettività di $g$
Basti considerare
$$f: \mathbf{N} \rightarrow \mathbf{Z}, f\left(x\right) = x - 1$$
$$g: \mathbf{N} \rightarrow \mathbf{N}, g\left(z\right) = z$$
Basti considerare $y = -2$
$$\textrm{Immagine}_f \cup \textrm{Immagine}_g = \left[-1, +\infty\right) \cup \left[0, +\infty\right) = \left[-1, +\infty\right) \Longrightarrow -2 \in \mathbf{Z} \notin \left[-1, +\infty\right)$$
